# Analyse de base des fichiers concernant les lycées
objectif : faire un fichier pour lancer des analyses.

In [1]:
# import bibliothéques d'analyse de python
import pandas as pd
import matplotlib.pyplot as plt
from matplotlib.ticker import MaxNLocator
from matplotlib.ticker import FuncFormatter
import seaborn as sns
import numpy as np

import fonctions.perso_stats as perso_stats

In [2]:
ips_lycee = pd.read_csv('donnee/fr-en-ips-lycees-ap2022.csv',sep=';')
va_lycee_gen = pd.read_csv('donnee/fr-en-indicateurs-de-resultat-des-lycees-denseignement-general-et-technologique.csv',sep=';',low_memory=False)
va_lycee_pro = pd.read_csv('donnee/fr-en-indicateurs-de-resultat-des-lycees-denseignement-professionnels.csv',sep=';',low_memory=False)

## affichage des fichiers bruts

In [3]:
ips_lycee.head(1)

Rentrée scolaire Académie Code du département Département       UAI  \
0        2022-2023     LYON                 001         AIN  0010021T   

                               Nom de l'établissment Code INSEE de la commune  \
0  LYCEE PROFESSIONNEL MARCELLE PARDE LYCEE DES M...                    01053   

  Nom de la commune Secteur Type de lycée  Effectifs voie GT  \
0   BOURG EN BRESSE  public            LP                NaN   

   Effectifs voie PRO  Effectifs Ensemble GT-PRO  IPS voie GT  IPS voie PRO  \
0               520.0                      520.0          NaN          86.5   

   IPS Ensemble GT-PRO  Ecart-type de l'IPS voie GT  \
0                 86.5                          NaN   

   Ecart-type de l'IPS voie PRO  
0                          24.3

In [4]:
ips_lycee.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3598 entries, 0 to 3597
Data columns (total 18 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Rentrée scolaire              3598 non-null   object 
 1   Académie                      3598 non-null   object 
 2   Code du département           3598 non-null   object 
 3   Département                   3598 non-null   object 
 4   UAI                           3598 non-null   object 
 5   Nom de l'établissment         3598 non-null   object 
 6   Code INSEE de la commune      3598 non-null   object 
 7   Nom de la commune             3598 non-null   object 
 8   Secteur                       3598 non-null   object 
 9   Type de lycée                 3598 non-null   object 
 10  Effectifs voie GT             2464 non-null   float64
 11  Effectifs voie PRO            2075 non-null   float64
 12  Effectifs Ensemble GT-PRO     3598 non-null   float64
 13  IPS

Le fichier fr-en-ips-lycees-ap2022.csv contient 18 colonnes<br>
<ul>
  <li>10 colonnes pour decriere le lycée (adresse, type de lycée,...)</li>
  <li>une colonne qui indique la rentrée au format XXXX-YYYY</li>
  <li>3 colonnes qui décrivent les effectifs général, professionnel et totaux</li>
  <li>5 colonnes pour décrire l'IPS selon le cursus général ou pro, ainsi que l'ecart-type</li>
</ul>

On va créer un carnet d'adresse avec le numero uai à partir de ce fichier.<br>
On va reformater la colonne rentrée scolaire (passée de XXXX-YYYY à XXXX)

In [5]:
va_lycee_gen.head(1)

Etablissement  Annee   Ville       UAI  \
0  LYCEE XAVIER BICHAT (GENERAL ET TECHNO.)   2021  NANTUA  0010032E   

  Code commune Academie Departement Secteur  Presents - L  Presents - ES  ...  \
0        01269     LYON         AIN  public           NaN            NaN  ...   

   Taux de mentions - Gnle  Valeur ajoutée du taux de mentions - Gnle  \
0                     70.0                                         -7   

   Nombre de mentions TB avec félicitations - G  \
0                                           NaN   

   Nombre de mentions TB sans félicitations - G  Nombre de mentions B - G  \
0                                           NaN                       NaN   

   Nombre de mentions AB - G  Nombre de mentions TB avec félicitations - T  \
0                        NaN                                           NaN   

   Nombre de mentions TB sans félicitations - T  Nombre de mentions B - T  \
0                                           NaN                       NaN   

   Nombre de mentions AB - T  
0                        NaN  

[1 rows x 145 columns]

In [6]:
va_lycee_gen.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27808 entries, 0 to 27807
Columns: 145 entries, Etablissement to Nombre de mentions AB - T
dtypes: float64(80), int64(3), object(62)
memory usage: 30.8+ MB


Le fichier fr-en-indicateurs-de-resultat-des-lycees-denseignement-general-et-technologique.csv contient les 145 colonnes suivantes :

In [7]:
for a in va_lycee_gen.columns :
    print(a)

Etablissement
Annee
Ville
UAI
Code commune
Academie
Departement
Secteur
Presents - L
Presents - ES
Presents - S
Presents - STG
Presents - STI2D
Presents - STD2A
Presents - STMG
Presents - STI
Presents - STL
Presents - ST2S
Presents - TMD
Presents - STHR
Presents - Toutes series
Taux de reussite - L
Taux de reussite - ES
Taux de reussite - S
Taux de reussite - STG
Taux de reussite - STI2D
Taux de reussite - STD2A
Taux de reussite - STMG
Taux de reussite - STI
Taux de reussite - STL
Taux de reussite - ST2S
Taux de reussite - TMD
Taux de reussite - STHR
Taux de reussite - Toutes series
Taux de reussite attendu acad - L
Taux de reussite attendu acad - ES
Taux de reussite attendu acad - S
Taux de reussite attendu acad - STG
Taux de reussite attendu acad - STI2D
Taux de reussite attendu acad - STD2A
Taux de reussite attendu acad - STMG
Taux de reussite attendu acad - STI
Taux de reussite attendu acad - STL
Taux de reussite attendu acad - ST2S
Taux de reussite attendu acad - TMD
Taux de reuss

Le fichier fr-en-indicateurs-de-resultat-des-lycees-denseignement-general-et-technologique.csv contient 145 colonnes<br>
<ul>
  <li>une colonne annee qui est l'année de la rentré scolaire</li>
  <li>11 colonne pour décrire l'etablissement (adresse, secteur public ou privé, numéro uai)</li>
  <li>pour chaque serie:
    <ul>
      <li>une colonne effectif_present qui donne le nombre de canditats qui pqsse le bac</li>
      <li>les taux brut de réussite</li>
      <li>les taux de réussite attendu en france, dans l'academie</li>
      <li>les taux de mention brut et attendu</li>
      <li>la valeur ajoutée pour le taux de réussite et la valeur ajouté pour le taux mention</li>
    </ul>
    </li>
    <li>8 colonnes qui donne le nombre de mention par college toute serie confondu</li>
</ul>
Alors pour limiter le nombre de colonnes : création d'une colonne serie et fusion de toutes les colonnes des series.<br>


In [8]:
va_lycee_pro.head(1)

Etablissement       UAI  Annee             Ville  \
0  LYCEE PROFESSIONNEL SAINTE JEANNE D'ARC  0592928M   2019  AULNOYE AYMERIES   

  Code commune Academie Departement             Secteur  \
0        59033    LILLE        NORD  privé sous contrat   

   Presents - Specialites pluri-technologiques de la production  \
0                                                NaN              

   Presents - Transformations  ...  \
0                         5.0  ...   

   Valeur ajoutee du taux de mentions - Specialites plurivalentes des services  \
0                                                NaN                             

   Valeur ajoutee du taux de mentions - Echanges et gestion  \
0                                                NaN          

   Valeur ajoutee du taux de mentions - Communication et information  \
0                                                NaN                   

   Valeur ajoutee du taux de mentions - Services aux personnes  \
0                                                  3             

   Valeur ajoutee du taux de mentions - Services a la collectivite  \
0                                                NaN                 

   Valeur ajoutee du taux de mentions - Services  \
0                                              3   

   Valeur ajoutee du taux de mentions - Toutes series  \
0                                                5.0    

   Nombre de mentions TB sans félicitations - P  Nombre de mentions B - P  \
0                                           NaN                       NaN   

   Nombre de mentions AB - P  
0                        NaN  

[1 rows x 130 columns]

In [9]:
va_lycee_pro.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24236 entries, 0 to 24235
Columns: 130 entries, Etablissement to Nombre de mentions AB - P
dtypes: float64(62), int64(3), object(65)
memory usage: 24.0+ MB


Le fichier fr-en-indicateurs-de-resultat-des-lycees-denseignement-professionnels.csv contient les 130 colonnes suivantes :

In [10]:
for a in va_lycee_pro.columns :
    print(a)

Etablissement
UAI
Annee
Ville
Code commune
Academie
Departement
Secteur
Presents - Specialites pluri-technologiques de la production
Presents - Transformations
Presents - Genie civil construction bois
Presents - Materiaux souples
Presents - Mecanique electricite electronique
Presents - Production
Presents - Specialites plurivalentes des services
Presents - Echanges et gestion
Presents - Communication et information
Presents - Services aux personnes
Presents - Services a la collectivite
Presents - Services
Presents - Toutes series
Taux de reussite - Specialites pluri-technologiques de la production
Taux de reussite - Transformations
Taux de reussite - Genie civil construction bois
Taux de reussite - Materiaux souples
Taux de reussite - Mecanique electricite electronique
Taux de reussite - Production
Taux de reussite - Specialites plurivalentes des services
Taux de reussite - Echanges et gestion
Taux de reussite - Communication et information
Taux de reussite - Services aux personnes
Tau

Le fichier contient les éléments suivants :<br>
<ul>
  <li>Une colonne annee avec l'année scolaire</li>
  <li>9 colonnes qui décrivent le lycée (adresse, secteur public ou privé, uai, ...)</li>
  <li>3 colonnes pour les mentions (pas de mention très bien avec félicitation par rapport au lycée général)</li>
  <li>pour chaque spécialité professionnelle :
    <ul>
      <li>une colonne effectif qui dinne le nombre de candidat diplome preparé</li>
      <li>un taux brut de réussite</li>
      <li>un taux de réussite attendu en france</li>
      <li>un taux de réussite attendu dans l'academie</li>
      <li>un taux brut de mention</li>
      <li>un taux de mention attendu</li>
      <li>la valeur ajouté du à l'etablissement au taux de réussite</li>
      <li>la valeur ajouté du à l établissement au taux de mention</li>
      <li>la valeur ajouté du à l'etablissement au taux d'accès (le lycée garde ces élèves au cours de la formation)</li>
    </ul>
  </li>
  <li>les taux d'accès de l'etablissement</li>
</ul>

Pour l'instant, je me concentre sur les différentes series, et l'IPS des établissements.<br>
Pour cela je vais créer comme pour le lycée général une colonne serie et fussionner les différente colonnes communes.<br>

In [11]:
adresse_lycee = ips_lycee[['UAI','Académie','Département','Code du département',"Nom de l'établissment",'Code INSEE de la commune','Nom de la commune','Type de lycée']].groupby(by='UAI',group_keys=False).apply(lambda x :x)

affichage des données uniques de la colonne Rentrée scolaire du fichier IPS :

In [12]:
print(ips_lycee['Rentrée scolaire'].unique())

['2022-2023']


affichage élements unique après changement de la colonne Rentrée scolaire du fichier IPS

In [13]:
ips_lycee['Rentrée scolaire'] = ips_lycee['Rentrée scolaire'].str[:4].astype(int)
print(ips_lycee['Rentrée scolaire'].unique())

[2022]


In [14]:
ips_lycee.drop(columns=['Secteur','Académie',
'Code du département','Département',"Nom de l'établissment",
'Code INSEE de la commune','Nom de la commune','Type de lycée',
'Effectifs voie GT','Effectifs voie PRO','Effectifs Ensemble GT-PRO'],inplace=True)

In [15]:
va_lycee_gen.rename(columns={'Annee':'Rentrée scolaire'},inplace=True)

In [16]:
# enlevé les resneignement type adresse
va_lycee_gen.drop(columns=['Code commune','Ville','Academie','Departement','Etablissement','Region','Libelle departement','Code region','Code departement'],inplace=True)

In [17]:
# faire une liste des series dans le fichier va_lycee_gen
liste_serie_gen = []

for nom_colonne in va_lycee_gen.columns:
    if nom_colonne.startswith('Presents'):
        liste_serie_gen.append(nom_colonne[8:])

In [18]:
# faire un dataframe avec serie

lycee_gen = pd.DataFrame(columns=['Rentrée scolaire','UAI','Secteur','Serie','Type de diplome gen pro','Presents'
                                    ,'Taux de reussite','Taux de reussite attendu acad','Taux de reussite attendu france'
                                    ,'Taux de mentions','Taux de mentions attendu'
                                    ,"Valeur ajoutee du taux de reussite"
                                    ,"Valeur ajoutee du taux de mentions"])
for serie in  liste_serie_gen: 
    liste_colonne = []
    colonne_mention =[]
    dico_serie = {'Presents'+serie:'Presents'
                ,'Taux de reussite'+serie:'Taux de reussite'
                ,'Taux de reussite attendu acad'+serie:'Taux de reussite attendu acad'
                ,'Taux de reussite attendu france'+serie:'Taux de reussite attendu france'
                ,'Taux de mentions'+serie:'Taux de mentions'
                ,'Taux de mentions attendu'+serie:'Taux de mentions attendu'
                ,"Valeur ajoutee du taux de reussite"+serie:"Valeur ajoutee du taux de reussite"
                ,"Valeur ajoutee du taux de mentions"+serie:"Valeur ajoutee du taux de mentions"
                }

    for colonne in va_lycee_gen.columns:
        if colonne.endswith(serie):
            liste_colonne.append(colonne)
        if colonne.startswith('nombre_de_mentions'):
            colonne_mention.append(colonne)
    liste_colonne +=['Rentrée scolaire','Secteur','UAI']
    temps_data = va_lycee_gen[~va_lycee_gen['Presents'+serie].isna()][liste_colonne].copy()
    temps_data['Serie']= serie
    temps_data['Type de diplome gen pro'] = 'gen'
    temps_data.rename(columns=dico_serie,inplace=True)
    lycee_gen = pd.concat([lycee_gen,temps_data])


Le fichier lycee gen après changement :

In [19]:
lycee_gen.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 120203 entries, 468 to 27807
Data columns (total 15 columns):
 #   Column                                     Non-Null Count   Dtype  
---  ------                                     --------------   -----  
 0   Rentrée scolaire                           120203 non-null  object 
 1   UAI                                        120203 non-null  object 
 2   Secteur                                    120203 non-null  object 
 3   Serie                                      120203 non-null  object 
 4   Type de diplome gen pro                    120203 non-null  object 
 5   Presents                                   120203 non-null  float64
 6   Taux de reussite                           120203 non-null  float64
 7   Taux de reussite attendu acad              31561 non-null   float64
 8   Taux de reussite attendu france            85271 non-null   object 
 9   Taux de mentions                           66106 non-null   float64
 10  Taux de

In [20]:
va_lycee_pro.rename(columns={'Annee':'Rentrée scolaire'},inplace=True)

In [21]:
va_lycee_pro.drop(columns=['Etablissement','Ville','Code commune','Academie','Departement'],inplace=True)

In [22]:
#listez les serie générale et technologique
liste_serie_pro = []
for nom_colonne in va_lycee_pro.columns:
    if nom_colonne.startswith('Presents'):
        liste_serie_pro.append(nom_colonne[8:])

In [23]:
# faire un dataframe avec serie

lycee_pro = pd.DataFrame(columns=['Rentrée scolaire','UAI','Secteur','Serie','Type de diplome gen pro','Presents'
                                    ,'Taux de reussite','Taux de reussite attendu acad','Taux de reussite attendu france'
                                    ,'Taux de mentions','Taux de mentions attendu'
                                    ,'Valeur ajoutee du taux de reussite',"Valeur ajoutee du taux de mentions"])
for serie in  liste_serie_pro: 
    liste_colonne = []
    dico_serie = {'Presents'+serie:'Presents'
                ,'Taux de reussite'+serie:'Taux de reussite'
                ,'Taux de reussite attendu acad'+serie:'Taux de reussite attendu acad'
                ,'Taux de reussite attendu france'+serie:'Taux de reussite attendu france'
                ,'Taux de mentions'+serie:'Taux de mentions'
                ,'Taux de mentions attendu'+serie:'Taux de mentions attendu'
                ,'Valeur ajoutee du taux de reussite'+serie:'Valeur ajoutee du taux de reussite'
                ,"Valeur ajoutee du taux de mentions"+serie:"Valeur ajoutee du taux de mentions"
                }
    for colonne in va_lycee_pro.columns:
        if colonne.endswith(serie):
            liste_colonne.append(colonne)
    liste_colonne +=['Rentrée scolaire','Secteur','UAI']
    temps_data = va_lycee_pro[~va_lycee_pro['Presents'+serie].isna()][liste_colonne].copy()
    temps_data['Serie']= serie
    temps_data['Type de diplome gen pro'] = 'pro'
    temps_data.rename(columns=dico_serie,inplace=True)
    lycee_pro = pd.concat([lycee_pro,temps_data])

Le Dataframe lycée pro après changement :

In [24]:
lycee_pro.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 102793 entries, 36 to 24235
Data columns (total 13 columns):
 #   Column                              Non-Null Count   Dtype  
---  ------                              --------------   -----  
 0   Rentrée scolaire                    102793 non-null  object 
 1   UAI                                 102793 non-null  object 
 2   Secteur                             102793 non-null  object 
 3   Serie                               102793 non-null  object 
 4   Type de diplome gen pro             102793 non-null  object 
 5   Presents                            102793 non-null  float64
 6   Taux de reussite                    102793 non-null  float64
 7   Taux de reussite attendu acad       8124 non-null    float64
 8   Taux de reussite attendu france     62621 non-null   object 
 9   Taux de mentions                    65827 non-null   float64
 10  Taux de mentions attendu            29109 non-null   object 
 11  Valeur ajoutee du taux de 

In [25]:
#merger les fichiers ensembles
fichier_lycee = pd.concat([lycee_gen,lycee_pro])
fichier_lycee = pd.merge(fichier_lycee,ips_lycee, on=['UAI','Rentrée scolaire'],how='left')
fichier_lycee = pd.merge(fichier_lycee,adresse_lycee, on='UAI',how='left')
#pas mal de colonne voir pour diminuer celle-ci ou pas


In [26]:
fichier_lycee['Secteur'].replace({'public':'PUBLIC','privé sous contrat':'PRIVE'},inplace=True)

Le fichier utilisé pour analyse :

In [27]:
fichier_lycee.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 222996 entries, 0 to 222995
Data columns (total 27 columns):
 #   Column                                     Non-Null Count   Dtype  
---  ------                                     --------------   -----  
 0   Rentrée scolaire                           222996 non-null  object 
 1   UAI                                        222996 non-null  object 
 2   Secteur                                    222996 non-null  object 
 3   Serie                                      222996 non-null  object 
 4   Type de diplome gen pro                    222996 non-null  object 
 5   Presents                                   222996 non-null  float64
 6   Taux de reussite                           222996 non-null  float64
 7   Taux de reussite attendu acad              39685 non-null   float64
 8   Taux de reussite attendu france            147892 non-null  object 
 9   Taux de mentions                           131933 non-null  float64
 10  Taux de 

In [28]:
fichier_lycee.to_csv('donnee_temp/lycee.csv',index=False)